app/controllers/application_controller.rb

```diff
@@ -3,8 +3,148 @@
 class ApplicationController < Sinatra::Base
 
   configure do
+    enable :sessions
+    set :session_secret, "password_security"
+  end
     set :public_folder, 'public'
     set :views, 'app/views'
+
+    get '/' do
+      erb :index
+    end
+
+    def logged_in?
+      !!session[:user_id]
+    end
+
+    get '/signup' do
+      if self.logged_in?
+          redirect "/tweets"
+      else
+        erb :'/users/create_user'
+      end
+    end
+
+    post '/signup' do
+      if params[:username]=="" || params[:email]==""|| params[:password]==""
```

Comment:
I think you could use [ActiveRecord validations](http://guides.rubyonrails.org/active_record_validations.html) (e.g. `validates_presence_of :username`) to perform these types of checks more easily.